src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+        if `X(e) \neq 0` else it equals `Y(e)`.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: OrientedMatroid.options.display='vector'
+            sage: M = OrientedMatroid([[0],[1],[-1]], key='vector')
+            sage: E1 = M.elements()[0]; E2 = M.elements()[1]
+            sage: E1.composition(E2)
+            (1)
+            sage: E2.composition(E1)
+            (1)
+            sage: E1.composition(E2) == E2.composition(E1)
+            True
+        """
+        p = []
+        n = []
+        z = []
+        for e in self.groundset():
+            x = self(e)
+            # If x is non-zero, keep its value
+            if x == 1:
+                p.append(e)
+            elif x == -1:
+                n.append(e)
+            else:
+                # else grab the value of the other
+                x = other(e)
+                if x == 1:
+                    p.append(e)
+                elif x == -1:
+                    n.append(e)
+                else:
+                    z.append(e)
+        return type(self)(self.parent(), positives=p, negatives=n, zeros=z)
+
+    def separation_set(self, other):
+        r"""
+        Return the separation set between two elements.
+
+        The separation set of two elements `X` and `Y`
+        is given by `S(X, Y) = \left\{e \mid X(e) = -Y(e) \neq 0 \right\}`
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1.separation_set(E2)
+            {0, 2}
+        """
+        return self.positives().intersection(other.negatives()).union(self.negatives().intersection(other.positives()))
+
+    def reorientation(self, change_set):
+        r"""
+        Return the reorientation by a set.
+
+        The reorientation of `X` by some `A \subseteq E` is
+        the signed subset (covector) given by `{}_{-A}X` where
+        `{}_{-A}X^+ = (X^+ \backslash A) \cup (X^- \cap A)` and similarly for
+        `{}_{-A}X^-`.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]; E
+            (1,0,-1)
+            sage: E.reorientation(2)
+            (1,0,1)
+            sage: E.reorientation([0,1,2])
+            (-1,0,1)
+        """
+        if change_set in self.groundset():
+            change_set = set([change_set])
+        else:
+            change_set = set(change_set)
+
+        # ensure every elt is in the groundset
+        for i in change_set:
+            if i not in self.groundset():
```

Comment:
```suggestion
            if i not in self._g:
```
and similar to the above changes to this method and to other methods.